src/classes/OPP_PrimaryContactRoleMerge.cls

```diff
@@ -0,0 +1,256 @@
+    * @description The Duplicate Primary Opportunity Contact Roles Wrapper. It contains the logic that loads  
+    * and manages the duplicate Primary OCRs and the related Opportunities.
+    ********************************************************************************************************/
+    public class DuplicatePrimaryOCRs {
+        private Map<Id, Opportunity> opportunityRecords;
+        private Map<Id, OpportunityContactRole> ocrRecords;
+
+        private Map<Id, Opportunity> opportunityRecordsToUpdate;
+        private Map<Id, OpportunityContactRole> ocrRecordsToUpdate;
+        private Map<Id, OpportunityContactRole> ocrRecordsToDelete;
+
+        /*********************************************************************************************************
+        * @description The Constructor of DuplicatePrimaryOCRs.
+        * @param List<AggregateResult> The duplicate primary OCRs Aggregate Result. 
+        */
+        public DuplicatePrimaryOCRs(List<AggregateResult> primaryOCRsAggregateResult) {
+            opportunityRecords = new Map<Id, Opportunity>(loadOpportunities(primaryOCRsAggregateResult));
+            ocrRecords = new Map<Id, OpportunityContactRole>(loadOpportunityContactRoles(opportunityRecords.values()));
+            opportunityRecordsToUpdate = new Map<Id, Opportunity>();
+            ocrRecordsToUpdate = new Map<Id, OpportunityContactRole>();
+            ocrRecordsToDelete = new Map<Id, OpportunityContactRole>();
+        }
+
+        /*******************************************************************************************************
+        * @description Gets the Opportunities of the duplicate primary OCRs. 
+        * @return List<Opportunity> The Opportunities of the duplicate primary OCRs.
+        ********************************************************************************************************/
+        public List<Opportunity> getOpportunities() {
+            return opportunityRecords.values();
+        }
+
+        /*******************************************************************************************************
+        * @description Sets the OCR in paramater as Primary OCR.
+        * @param OpportunityContactRole the OCR to set as primary.
+        * @return void.
+        ********************************************************************************************************/
+        public void makePrimary(OpportunityContactRole inputOCR) {
+            OpportunityContactRole ocrToUpdate = ocrRecords.get(inputOCR.Id);
+            ocrToUpdate.IsPrimary = true;
+            ocrRecordsToUpdate.put(ocrToUpdate.Id, ocrToUpdate);
+
+            Opportunity opportunityToUpdate = opportunityRecords.get(inputOCR.OpportunityId);
+            if (opportunityToUpdate.Primary_Contact__c != inputOCR.ContactId) {
+                opportunityToUpdate.Primary_Contact__c = inputOCR.ContactId;
```

Comment:
Nic is verifying the logic on this whether we should be updating the opp or if the contact specified on the opp should win.